checks/Dependencies.scala

```diff
@@ -0,0 +1,92 @@
+import scala.util.Try
+import scala.util.matching.Regex
+import upickle.default.*
+import coursier.graph.DependencyTree
+import Dependencies.*
+import coursier.*
+import scala.annotation.tailrec
+
+object Dependencies:
+  case class Version(major: Int, minor: Int, patch: Int, suffix: Option[String] = None) extends Ordered[Version] derives ReadWriter:
+    def compare(that: Version): Int =
+      Ordering[(Int, Int, Int, Option[String])].compare((this.major, this.minor, this.patch, this.suffix), (that.major, that.minor, that.patch, that.suffix))
+    override def toString: String = s"$major.$minor.$patch${suffix.fold("")(s => s"-$s")}"
+    def getDiff(that: Version): VersionDiff = Version.compareVersions(this, that)
+
+  object Version:
+      /**
+     * Compares two versions and returns the type of update that is required to go from oldVersion to newVersion.
+     * Takes into account that versions may be rolled back. For example:
+     *   - 1.0.1 -> 1.0.0 is a patch update
```

Comment:
Assuming that the library authors are very strictly following early semver. In my experience, not all libraries/maintainers follow this scheme: they will add small features in patches and only bump minors for more significant changes. Furthermore, as far as I know the tooling (i.e. MiMa) is not currently able to programmatically enforce strict early-semver, so this is relying on maintainers best efforts.

Maybe that's okay for toolkit, just pointing it out.

Reply:
I want to solve it with the list of exceptions. If it's impossible to follow that, then ok, but let's have a list showing where it breaks.